src/ts/forwarders/proxyAccountFowarder.ts

```diff
@@ -79,38 +134,146 @@ export class ProxyAccountForwarder extends Forwarder<ProxyAccountCallData> {
    * an Ethereum Transaction's data field.
    * @param params Forward Parameters
    */
-  public async encodeSignedMetaTransaction(
+  protected async encodeSignedMetaTransaction(
     params: ForwardParams
-  ): Promise<string> {
+  ): Promise<MinimalTx> {
     const proxyAccount = new ProxyAccountFactory(this.signer).attach(params.to);
+    let callData;
+
+    if (params.callType == CallType.DELEGATE) {
+      callData = proxyAccount.interface.functions.delegate.encode([
+        { target: params.target, value: params.value, callData: params.data },
+        params.replayProtection,
+        params.replayProtectionAuthority,
+        params.signature,
+      ]);
+    } else {
+      // We assume it is a CALL - safe default.
+      callData = proxyAccount.interface.functions.forward.encode([
+        { target: params.target, value: params.value, callData: params.data },
+        params.replayProtection,
+        params.replayProtectionAuthority,
+        params.signature,
+      ]);
+    }
 
-    return proxyAccount.interface.functions.forward.encode([
-      params.target,
-      params.value,
-      params.data,
-      params.replayProtection,
-      params.replayProtectionAuthority,
-      params.signature,
+    return { to: params.to, data: callData, value: params.value };
+  }
+
+  /**
+   * Batches a list of transactions into a single meta-transaction.
+   * It supports both meta-transactions & meta-deployment.
+   * @param dataList List of meta-transactions to batch
+   */
+  public async signAndEncodeBatchTransaction(
```

Comment:
This seems to share a lot of code with the relayHub function

Reply:
ya. if we are tidying up API we could address it there potentially. there are some subtle differences. we would need:
- modify getEncodedCallData() to include a .BATCH version. although the fields we encode are different for both approaches. 
- encoding the batch function would need its own abstract function, implemented in both forwarders. 
- the initial loop to set defaults for the optional fields is also slightly different (proxy has more optional fields).

Reply:
Ok, lets address that stuff in a big "tidy up" PR